backend/modules/simulation/agents/persona_agent.py

```
@@ -248,169 +305,187 @@ def get_persona_knowledge(query: str) -> str:
-            escaped_system_prompt = self._get_system_prompt()
-            return ChatPromptTemplate.from_messages([
-                ("system", escaped_system_prompt),
-                MessagesPlaceholder(variable_name="chat_history"),
-                ("human", "{input}"),
-                MessagesPlaceholder(variable_name="agent_scratchpad")
-            ])
-        # Default persona prompt with history/tools when no custom prompt provided
+        """Create persona-specific prompt template (no scene context — used in non-streaming path)."""
+        return self._create_persona_prompt_with_attempt(attempt_number=1, scene_context=None)
+    
+    def _create_persona_prompt_with_attempt(self, attempt_number: int, scene_context: Dict[str, Any] | None = None) -> ChatPromptTemplate:
```

Ruff (0.15.1)

[warning] Unused method argument: `attempt_number`

(ARG002)